### PR TITLE
Code Exposure Initial Commit

### PR DESCRIPTION
**Pull request by sjaws**
Code Exposure initial commit

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/Code Exposure.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+- `Who is monitoring the logs/alerts, receiving them and acting upon each?`
+- `Who gets notified when an alert is discovered?`
+- `When do public relations and legal get involved in the process?`
+- `When would you reach out to AWS Support for help?`
+
+## Analysis
+It is highly recommended to export logs to a security incident event management (SIEM) solution (such as Splunk, ELK stack, etc.) to aid in viewing and analyzing a variety of logs for a more complete attack timeline analysis.
+
+### CloudTrail: Public S3 Bucket
+By default, CloudTrail logs API calls that were made in the last 90 days, but not log requests made to objects. You can see bucket-level events on the CloudTrail console. However, you can't view data events (Amazon S3 object-level calls) there—you must parse or query CloudTrail logs for them.
```
Comment (kihakuni) [inline docs/Code Exposure.md]: "there—you must parse or query CloudTrail logs for them. "

Recommend changing this to "by default--you must enable object level logging before those events appear in CloudTrail."

File: docs/Code Exposure.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+### CloudTrail: Public S3 Bucket
+By default, CloudTrail logs API calls that were made in the last 90 days, but not log requests made to objects. You can see bucket-level events on the CloudTrail console. However, you can't view data events (Amazon S3 object-level calls) there—you must parse or query CloudTrail logs for them. 
+
+1. Navigate to your [CloudTrail Dashboard](https://console.aws.amazon.com/cloudtrail)
+1. In the left-hand margin select `Event History`
+1. In the drop-down change from `Read-Only` to `Event Name`
+1. Review CloudTrail logs for the eventnames `GetPublicAccessBlock` and `DeletePublicAccessBlock`
+
+### CloudTrail: Public S3 Object
+You can also get CloudTrail logs for object-level Amazon S3 actions. To do this, enable data events for your S3 bucket or all buckets in your account. When an object-level action occurs in your account, CloudTrail evaluates your trail settings. If the event matches the object that you specified in a trail, the event is logged.
```
Comment (kihakuni) [inline docs/Code Exposure.md]: Recommend adding the process to enable events OR linking "enable data events" to the process in our documentation so it's clear how to do so

File: docs/Code Exposure.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+Approver: `Approver Name`  
+Last Date Approved:   
+
+## Executive Summary
+This playbook outlines the process to identify owners of code resources, determine how they were exposed, who may have accessed the exposed code, determine impact of exposure, required remediation actions, and determine root cause of code exposure.
+
+## Potential Indicators of Compromise
+- DLP Alerts
+- Known code publication or exposure
+- Suspicious CloudTrail logs
```
Comment (kihakuni) [inline docs/Code Exposure.md]: I see DLP and code publication mentioned here, but not elsewhere in the document? 

Did you want to focus on S3, or include these in procedures as well for a more complete document?

File: docs/Code Exposure.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+14. [**RECOVERY**] Perform recovery procedures as appropriate
+
+***The response steps follow the Incident Response Life Cycle from [NIST Special Publication 800-61r2 Computer Security Incident Handling Guide](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-61r2.pdf)  
+
+![Image](/images/nist_life_cycle.png)***
+
+### Incident Classification & Handling
+* **Tactics, techniques, and procedures**: Data Exfiltration, AWS Service Abuse
+* **Category**: Data Loss
+* **Resource**: CodeCommit, S3, EC2
```
Comment (kihakuni) [inline docs/Code Exposure.md:90]: I see CodeCommit mentioned here. Do we have incident response or review procedures for CodeCommit? 

Document focuses on S3, but I didn't see anything related to reviewing CodeCommit events beyond asset inventory

File: docs/Code Exposure.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+* **Indicators**: Cyber Threat Intelligence, Third Party Notice, CloudWatch Metrics
+* **Log Sources**: DNS Logs, VPC Flow Logs, CloudTrail, CloudWatch
```
Comment (kihakuni) [inline docs/Code Exposure.md:92]: Is this accurate? Or are you only covering Cloudtrail and VPC Flow Logs in this document?

File: docs/Code Exposure.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+## Detection
+
+### Logging and S3 Bucket Checks
+- Ensure CloudTrail is enabled in all regions: `./prowler -c check21`
+- Ensure CloudTrail log file validation is enabled: `./prowler -c check22`
+- Ensure S3 bucket access logging is enabled on the CloudTrail S3 bucket: `./prowler -c check26`
+- Ensure there are no S3 buckets open to Everyone or Any AWS user: `./prowler -c extra73`
+- Identify the resources in your organization and accounts; such as Amazon S3 buckets or IAM roles; that are shared with an external entity: `./prowler -c extra769`
+- Find resources exposed to the internet: `./prowler -g group17`
```
Comment (kihakuni) [inline docs/Code Exposure.md:160]: Do we have similar things we can look for security checks against CodeCommit? 
For example, this git project? https://aws.amazon.com/blogs/compute/discovering-sensitive-data-in-aws-codecommit-with-aws-lambda-2/

File: docs/Code Exposure.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+## Containment
+
+### S3 Block Public Access
+aws s3api put-public-access-block --bucket bucket-name-here --public-access-block-configuration "BlockPublicAcls=true,IgnorePublicAcls=true,BlockPublicPolicy=true,RestrictPublicBuckets=true"
+
+You can also review [Blocking public access to your Amazon S3 storage](https://aws.amazon.com/s3/features/block-public-access/) for additional details on blocking public S3 access across your account.
+
+## Eradication
+
```
Comment (kihakuni) [inline docs/Code Exposure.md:237]: Did you want to include processes/documents for EC2 and CodeCommit?

File: docs/Code Exposure.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+1. In the Bucket name list, choose the name of the bucket that you want to delete an object from.
+1. Choose the name of the object that you want to delete.
+1. To delete the current version of the object, choose Latest version, and choose the trash can icon.
+1. To delete a previous version of the object, choose Latest version, and choose the trash can icon beside the version that you want to delete.
+
+## Recovery
+Same procedures as those listed for Eradication
+
+## Preventative Actions
+
```
Comment (kihakuni) [inline docs/Code Exposure.md:262]: Any preventative actions for EC2 and CodeCommit?

**Review by kihakuni — CHANGES_REQUESTED** [review]

File: docs/Code Exposure.md (hunk 1 above)
Comment (sjaws) [reply docs/Code Exposure.md]: Added recommended change.

File: docs/Code Exposure.md (hunk 2 above)
Comment (sjaws) [reply docs/Code Exposure.md]: Added recommended link.

File: docs/Code Exposure.md (hunk 3 above)
Comment (sjaws) [reply docs/Code Exposure.md]: Added DLP and CloudTrail details.

File: docs/Code Exposure.md (hunk 4 above)
Comment (sjaws) [reply docs/Code Exposure.md:90]: Added content for CodeCommit and EC2.

File: docs/Code Exposure.md (hunk 5 above)
Comment (sjaws) [reply docs/Code Exposure.md:92]: Added content for additional resources and log sources.

File: docs/Code Exposure.md (hunk 6 above)
Comment (sjaws) [reply docs/Code Exposure.md:160]: While I am unaware of prowler checks for CodeCommit, I did add CodeCommit logging checks.

File: docs/Code Exposure.md (hunk 7 above)
Comment (sjaws) [reply docs/Code Exposure.md:237]: Added documentation for EC2 and CodeCommit.

File: docs/Code Exposure.md (hunk 8 above)
Comment (sjaws) [reply docs/Code Exposure.md:262]: Added documentation for EC2 and CodeCommit.

**Comment by sjaws** [comment]
Updated pull request.

**Review by kihakuni — APPROVED** [review]

**Pull request merged**